Unable to build `opensn` with `openmpi`

When using `openmpi` the code does not build. The error message seen is:
```
/path/mpicpp-lite/include/mpicpp-lite/impl/Group.h:125:19: error: incompatible integer to pointer conversion assigning to 'MPI_Group' (aka 'ompi_group_t *') from '(unnamed enum at /path/mpicpp-lite/include/mpicpp-lite/impl/Enums.h:5:1)'
    this->group = UNDEFINED;
                  ^~~~~~~~~
```

This is a problem in `mpicpp-lite`, see [issue](https://github.com/andrsd/mpicpp-lite/issues/21).

--- a/external/mpicpp-lite/impl/Group.h
+++ b/external/mpicpp-lite/impl/Group.h
@@ -122,7 +122,6 @@ inline void
 Group::free()
 {
     MPI_CHECK(MPI_Group_free(&this->group));
-    this->group = UNDEFINED;
 }
 
 inline int

